torch/csrc/jit/subgraph_matcher.cpp

```diff
@@ -21,24 +21,24 @@ class SubgraphMatcher {
    * is the same as in the corresponding matchGraph node, its type is the same,
    * and all nodes producing input-values also match.
    */
-  bool matchesSubgraphFromAnchorNode(const Node* anchor);
+  bool matchesSubgraphFromAnchorNode(Node* anchor);
 
   /** \brief Return match map for nodes. */
-  std::unordered_map<const Node*, const Node*> nodes_map() const {
+  std::unordered_map<const Node*, Node*> nodes_map() const {
     return nodes_map_;
   }
 
   /** \brief Return match map for values. */
-  std::unordered_map<const Value*, const Value*> values_map() const {
+  std::unordered_map<const Value*, Value*> values_map() const {
     return values_map_;
   }
 
  private:
-  bool matchValues(const Value* v1, const Value* v2);
-  bool matchNodes(const Node* n1, const Node* n2);
+  bool matchValues(const Value* v1, Value* v2);
```

Comment:
Do we really have to remove those? A non-const pointer can always be implicitly casted to a const pointer, so this shouldn't be necessary.

Reply:
Yes, we do.  The RHS Values/Nodes become the values in the mapping represented by the match.  If they are const here, then it propagates all the way through to the return values where the const_casts were.